packaging/test_plugin_unified_storage_tiering.py

```diff
@@ -1693,3 +1697,254 @@ def test_no_tiering_occurs_when_custom_violating_query_selects_correct_columns_i
+
+
+class test_executing_rules_as_rodsuser__issue_293(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        self.user1 = session.mkuser_and_return_session("rodsuser", "tolstoy", "tpass", lib.get_hostname())
+
+        self.tier0 = "ufs0"
+        self.tier1 = "ufs1"
+        self.tier0_time_in_seconds = 5
+
+        self.filename = "test_executing_rules_as_rodsuser__issue_293"
+        if not os.path.exists(self.filename):
+            lib.create_local_testfile(self.filename)
+
+        self.collection_path = self.user1.home_collection
+        self.object_path = "/".join([self.collection_path, self.filename])
+
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand(['iqdel', '-a'])
+
+            # Using HOSTNAME_2 here so that topology testing takes effect should we ever employ it in plugin testing.
+            lib.create_ufs_resource(admin_session, self.tier0, test.settings.HOSTNAME_2)
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier0, 'irods::storage_tiering::group', 'example_group', '0'])
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier0, 'irods::storage_tiering::time', str(self.tier0_time_in_seconds)])
+            admin_session.assert_icommand(
+                ['imeta', 'ls', '-R', self.tier0], 'STDOUT', [f'value: {self.tier0_time_in_seconds}', 'value: example_group'])
+
+            # Using HOSTNAME_3 here so that topology testing takes effect should we ever employ it in plugin testing.
+            lib.create_ufs_resource(admin_session, self.tier1, test.settings.HOSTNAME_3)
+            admin_session.assert_icommand(
+                ['imeta', 'add', '-R', self.tier1, 'irods::storage_tiering::group', 'example_group', '1'])
+            admin_session.assert_icommand(['imeta', 'ls', '-R', self.tier1], 'STDOUT', 'value: example_group')
+
+    @classmethod
+    def tearDownClass(self):
+        if os.path.exists(self.filename):
+            os.unlink(self.filename)
+
+        self.user1.__exit__()
+
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.run_icommand(['iadmin', 'rmuser', self.user1.username])
+
+            admin_session.run_icommand(['iqdel', '-a'])
+
+            admin_session.run_icommand(['iadmin', 'rmresc', self.tier0])
+            admin_session.run_icommand(['iadmin', 'rmresc', self.tier1])
+            admin_session.run_icommand(['iadmin', 'rum'])
+
+    def tearDown(self):
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.run_icommand(["ichmod", "-M", "own", admin_session.username, self.object_path])
+            admin_session.run_icommand(["irm", "-f", self.object_path])
+
+    def test_rule_invoked_by_rodsuser_directly_via_irule_fails(self):
+        with storage_tiering_configured():
+            IrodsController().restart(test_mode=True)
```

Comment:
The `.restart` instructions likely need to be tweaked for main since REP configs no longer take affect immediately.

Do the `storage_tiering_configured()` functions automatically restart/reload the server after leaving the scope?

Reply:
Good point. No, `storage_tiering_configured` does not restart the server.

Here's the thing: Tests are not passing on main right now because of #290. I'm thinking we should open an issue for this to fix tests for main *or* we can fix this as part of #290 because we will expect tests to pass once that is fixed. I can fix it here, but confirming that it works is contingent on #290 being fixed. I don't mind doing that, but the trade-off is that it will delay 4.3.3.1 because I'm not sure how long #290 will take.

Thoughts?

Reply:
That's right. Forgot this came up before.

Fixing this can wait. Don't want to delay 4.3.3.1.

Reply:
Created https://github.com/irods/irods_capability_storage_tiering/issues/300 (!) for updating these things. I don't think we need to add TODO comments because... there's a lot of these in this test suite.

Reply:
Yep, TODOs aren't required. We have an issue for it.

Reply:
Sounds good. Resolving